lib/typing/columns/columns.go

```diff
@@ -126,6 +127,10 @@ func (c *Columns) UpsertColumn(colName string, arg UpsertColumnArg) {
 			col.backfilled = *arg.Backfilled
 		}
 
+		if arg.StringPrecision != nil {
+			col.KindDetails.OptionalStringPrecision = arg.StringPrecision
```

Comment:
Since this has access to the existing column metadata, is this a better place to check that the new precision is greater than the previous?

Reply:
We already have guardrail in how NewLength is being set, I can add something here as well?

Reply:
My main concern was addressed on the other comment, but yeah it might be nice to add a backup check here too, just in case.

Reply:
Yeah I don't see why not. I'll add it

Reply:
I'm gonna open up another PR to add `error` to the signature for UpsertColumn so there's less diff